InteractionBase/src/vis/Menu.cpp

```diff
@@ -0,0 +1,245 @@
+{
+	//Move the cursor to original owner
+	target_->moveCursor();
+	//Shouldn't be destroyed
+	target_ = nullptr;
+	//Destroyed by DeclarativeItem
+	focusedItem_ = nullptr;
+	SAFE_DELETE_ITEM(textField_);
+	//Destroyed when the focused item is destroyed
+	if (!focusedItem() || focusedItem()->graphicsEffect() != selectedEffect_)
+		SAFE_DELETE(selectedEffect_);
+	selectedEffect_ = nullptr;
+}
+
+void Menu::initializeForms()
+{
+	auto nodeGrid = (new Visualization::DynamicGridFormElement())
+			->setSpacing(5, 5)
+			->setMajorAxis(Visualization::GridLayouter::ColumnMajor)
+			->setNodesGetter([](Visualization::Item* v)
+				{ auto self = static_cast<I*>(v);
+				  return self->currentNodes(); });
+	//First form: only the nodes
+	addForm(nodeGrid);
+	//Second form: the nodes and a text field
+	addForm((new Visualization::GridLayoutFormElement())
+			->setSpacing(5, 5)->setMargins(10)
+			->put(0, 0, item<Visualization::Text>(&I::textField_,
+									[](I* v) -> const Visualization::TextStyle* { return &v->style()->nameField(); }))
+									//[](I* v) { return v->style()->nameField(); }))
+			->put(0, 1, nodeGrid));
+}
+
+int Menu::determineForm()
+{
+	return hasTextField() ? 1 : 0;
+}
+
+bool Menu::hasTextField() const
+{
+	return true;
+}
+
+void Menu::selectItem(Visualization::Item *item)
+{
+	if (focusedItem() && focusedItem() != textField())
+	{
+		focusedItem()->setGraphicsEffect(nullptr);
+		selectedEffect_ = new QGraphicsColorizeEffect();
+	}
+	if (item && item != textField())
+		item->setGraphicsEffect(selectedEffect_);
+	if (item == textField())
+	{
+		item->moveCursor();
+		item->correspondingSceneCursor<Visualization::TextCursor>()->selectAll();
+	}
+	else moveCursor(Visualization::Item::MoveOnPosition, QPoint(0, 0));
+	focusedItem_ = item;
+}
+
+bool Menu::executeFocused()
+{
+	if (hasTextField() && focusedItem() == textField())
+		return onSelectText(textField()->text());
+	else if (focusedItem())
+		return onSelectNode(focusedItem()->node());
+	return false;
+}
+
+QPoint Menu::indexOf(Model::Node *node) const
+{
+	for (int col = 0; col < currentNodes_.size(); col++)
+		for (int row = 0; row < currentNodes_[col].size(); row++)
+			if (currentNodes_[col][row] == node)
+				return QPoint(col, row);
+	return QPoint(-1, -1);
+}
+
+QPoint Menu::correctCoordinates(QPoint point) const
+{
+	QPoint result = point;
+	if (point.x() < 0)
+		result.setX(currentNodes_.size() - 1);
+	if (point.x() >= currentNodes_.size())
+		result.setX(0);
+	if (point.y() < 0)
+		result.setY(currentNodes_[point.x()].size() - 1);
+	if (point.y() >= currentNodes_[result.x()].size())
+		result.setY(0);
+	return result;
+}
+
+void Menu::determineChildren()
+{
+	Super::determineChildren();
+	Item* defaultProxy{};
+	if (hasTextField())
+		setDefaultMoveCursorProxy(defaultProxy = textField_);
+	else
+		setDefaultMoveCursorProxy(defaultProxy = findVisualizationOf(currentNodes()[0][0]));
+	//TODO@cyril doing this in show doesn't work -> the grid does not have items yet
```

Comment:
You need to instruct the scene to focus the item *after* it has already created the children. Try this at the place you do show:
`target->scene()->addPostEventAction( new Interaction::SetCursorEvent(menuItem));`
where `menuItem` is the visualization corresponding to this Menu object.